src/client/main.ts

```diff
@@ -0,0 +1,280 @@
+import {channelPromise, tryServiceDispatch} from './connection';
+import {Context} from './context';
+import {IntentType} from './intents';
+import {APITopic, RaiseIntentPayload, SERVICE_IDENTITY} from './internal';
+
+/**
+ * This file was copied from the FDC3 v1 specification.
+ *
+ * Original file: https://github.com/FDC3/FDC3/blob/master/src/api/interface.ts
+ */
+
+// Re-export types/enums at top-level
+export * from './context';
+export * from './directory';
+export * from './intents';
+
+export enum OpenError {
+    AppNotFound = 'AppNotFound',
+    ErrorOnLaunch = 'ErrorOnLaunch',
+    AppTimeout = 'AppTimeout',
+    ResolverUnavailable = 'ResolverUnavailable'
+}
+
+export enum ResolveError {
+    NoAppsFound = 'NoAppsFound',
+    ResolverUnavailable = 'ResolverUnavailable',
+    ResolverTimeout = 'ResolverTimeout'
+}
+
+/**
+ * Intent descriptor
+ */
+export interface IntentMetadata {
+    name: string;
+    displayName: string;
+}
+
+/**
+ * An interface that relates an intent to apps
+ */
+export interface AppIntent {
+    intent: IntentMetadata;
+    apps: AppMetadata[];
+}
+
+/**
+ * App metadata is Desktop Agent specific - but should support a name property.
+ */
+export interface AppMetadata {
+    id: number;
+    name: string;
+    title: string;
+    manifest_url: string;
```

Comment:
Assume it's not our fault we're breaking from camelCase convention here?

Reply:
I just copied this from a different location, but probably comes from the runtime.

Reply:
This comes from the FDC3 App Directory specs.

Reply:
Thanks Phil! 👍